share/new_eds/nd.go

```diff
@@ -0,0 +1,30 @@
+package eds
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/celestiaorg/celestia-node/share"
+	"github.com/celestiaorg/celestia-node/share/shwap"
+)
+
+// NamespacedData extracts shares for a specific namespace from an EDS, considering
+// each row independently.
+func NamespacedData(
```

Comment:
Why this isn't a method from `Accessor`  but a standalone func? 🤔

Reply:
NamespacedData works using Accessor API and will be the same for all implementations. There is no reason to put it into interface now.